tic_tac_toe/game.py

```diff
@@ -87,18 +158,50 @@ def move(game, player, position):
     checks before the actual movement is done.
     After registering the movement it must check if the game is over.
     """
-    pass
+    
+    if get_next_turn(game) != player:
+        raise InvalidMovement('"{}" moves next'.format(get_next_turn(game)))
+    
+    if not _position_is_valid(position):
+        raise InvalidMovement('Position out of range.')
+        
+    
+    if _board_is_full(game['board']) or game['winner'] != None:
+        raise InvalidMovement('Game is over.')
+        
+    if not _position_is_empty_in_board(position, game['board']):
+        raise InvalidMovement('Position already taken.')
+    
+    if _position_is_empty_in_board(position, game['board']):
+        game['board'][position[0]][position[1]] = player
+        game['next_turn'] = _switch_player(game, player)
+    
+    if _check_winning_combinations(game['board'], player):
+        game['winner'] = player
+        raise GameOver('"{}" wins!'.format(player))
+    elif _board_is_full(game['board']):
+        raise GameOver('Game is tied!')
+
 
 
 def get_board_as_string(game):
     """
     Returns a string representation of the game board in the current state.
     """
-    pass
+    game = game['board']
+    
+    return '''
+{}  |  {}  |  {}
+--------------
+{}  |  {}  |  {}
+--------------
+{}  |  {}  |  {}
+'''.format(game[0][0], game[0][1], game[0][2], game[1][0], game[1][1], game[1][2], game[2][0], game[2][1], game[2][2])
```

Comment:
There's probably a more simplistic way to accomplish this formatting string. Also, watch your namespace. I'd advise you turn `game = game['board']` into `board = game['board']` for clarity.